### PR TITLE
Bugfix: buffered replication task is not correctly deleted

### PR DESCRIPTION
**Pull request by wxing1292**
solve #912

### CODE REVIEW

File: service/history/mutableStateBuilder.go [hunk 1]
```diff
@@ -255,7 +255,7 @@ type (
 		newBufferedEvents                *persistence.SerializedHistoryEventBatch
 		clearBufferedEvents              bool
 		newBufferedReplicationEventsInfo *persistence.BufferedReplicationTask
-		deleteBufferedReplicationEvent   *int64
+		deleteBufferedReplicationEvent   []int64
```
Comment (samarabbas) [inline service/history/mutableStateBuilder.go]: deleteBufferedReplicationEvents?

File: common/persistence/dataInterfaces.go [hunk 2]
```diff
@@ -574,22 +574,22 @@ type (
...
+		UpsertRequestCancelInfos       []*RequestCancelInfo
+		DeleteRequestCancelInfo        *int64
+		UpsertSignalInfos              []*SignalInfo
+		DeleteSignalInfo               *int64
+		UpsertSignalRequestedIDs       []string
+		DeleteSignalRequestedID        string
+		NewBufferedEvents              *SerializedHistoryEventBatch
+		ClearBufferedEvents            bool
+		NewBufferedReplicationTask     *BufferedReplicationTask
+		DeleteBufferedReplicationTasks []int64
```
Comment (samarabbas) [inline common/persistence/dataInterfaces.go]: I thought we only process single buffered event in each transaction.  What is the use case for processing more than one buffered replication task?

Comment (wxing1292) [reply common/persistence/dataInterfaces.go]: https://github.com/uber/cadence/blob/master/service/history/historyReplicator.go#L419

File: service/history/mutableStateBuilder.go (hunk 1 above)
Comment (wxing1292) [reply service/history/mutableStateBuilder.go]: sure

**Comment by wxing1292** [comment]
looks like for each buffered event, we do transaction to delete it and apply it, close this PR

**Pull request closed without merging**